src/components/Common/Sidebar.jsx

```diff
@@ -13,15 +11,13 @@ import {
 
 const Sidebar = ({ activekey }) => {
   // // initial state
-  const navigate = useNavigate();
+
   const dispatch = useDispatch();
   const user_id = localStorage.getItem('id');
   const role_id = localStorage.getItem('role_id');
   //Redux State
-  const { menuList, filterMenuList } = useSelector((state) => state?.sidebar);
-  const menuListData = menuList?.menu;
-  //local state
-  const [menuData, setMenuData] = useState(filterMenuList?.[0]);
+  const { menu } = useSelector((state) => state?.sidebar?.menuList || {});
```

Comment:
This approach could become problematic if we need more state variables in the future. The current setup assumes we only require menu from state.sidebar.menuList. If we anticipate needing additional state data later on, this structure might not scale well. It's worth reconsidering the design to ensure it can accommodate future changes smoothly.

Reply:
resolved